src/hu_stuff.c

```diff
@@ -523,6 +523,100 @@ static void HU_widget_build_speed(void);
 
 static hu_multiline_t *w_stats;
 
+typedef enum {
+  STATSFORMAT_RATIO,
+  STATSFORMAT_BOOLEAN,
+  STATSFORMAT_PERCENTAGE,
+  STATSFORMAT_REMAINING,
+  STATSFORMAT_COUNT,
+
+  NUM_STATSFORMATS
+} statsformat_t;
+
+static statsformat_t hud_stats_format;
+
+static int StatsFormatFunc_Ratio(
+  char *const str, const size_t str_size,
```

Comment:
I wonder if these function can be simplified more. Two quick examples (maybe there's a better approach than this):
- Use `hud_stringbuffer` and `sizeof(hud_stringbuffer)` directly, then pass `offset` to the function instead, even if it's zero
- `prefix_color` is always the same color, so maybe just use it directly

But maybe it's best to keep it generalized. The large number of inputs makes me wonder though...

Reply:
@ceski-1 is right. You can keep most of the strings constant and only fill the actual numbers using the function pointers. Strictly, these functions only need to get passed an array pointer, the size of that array, the count and the total count.

Reply:
For that matter, should I use a "function factory" like `r_draw.c:DRAW_COLUMN()`?

Reply:
> * `prefix_color` is always the same color, so maybe just use it directly

I added macro for that in ID24 SBARDEF:
```c
#define ORIG_S  "\x1b\x2f"
#define BRICK_S "\x1b\x30"
#define TAN_S   "\x1b\x31"
#define GRAY_S  "\x1b\x32"
#define GREEN_S "\x1b\x33"
#define BROWN_S "\x1b\x34"
#define GOLD_S  "\x1b\x35"
#define RED_S   "\x1b\x36"
#define BLUE_S  "\x1b\x37"

M_snprintf(string, sizeof(string), GRAY_S "%d " GREEN_S "FPS", fps);
```
I think we are doing double works with these HUD PRs.

> For that matter, should I use a "function factory" like `r_draw.c:DRAW_COLUMN()`?

No, please don't make it too complicated. Let's just reduce the number of parameters for these functions.